platforms/core-configuration/declarative-dsl-core/src/main/kotlin/org/gradle/internal/declarativedsl/analysis/PropertyAccessResolver.kt

```diff
@@ -73,8 +78,19 @@ class PropertyAccessResolverImpl(
     @Suppress("NestedBlockDepth")
     private
     fun AnalysisContext.doResolvePropertyAccessToObject(
-        propertyAccess: PropertyAccess
+        propertyAccess: PropertyAccess,
+        expectedType: DataTypeRef?
     ): ObjectOrigin? {
+        if (propertyAccess.receiver == null && expectedType != null && expectedType is DataTypeRef.Name) {
+            val dataType = schema.findTypeWithName(expectedType.fqName)
+            if (dataType is EnumClass) {
+                return dataType.entryNames.firstOrNull { it ==  propertyAccess.name}
```

Comment:
If there is no matching entry, this code should not return null immediately. The property resolution should go on to try and resolve the named reference to anything else.